supervisor_types.go

```diff
@@ -136,27 +136,34 @@ type Dimension struct {
 // SpatialDimensionSet is a unique set of druid datasource spatial dimensions.
 type SpatialDimensionSet []SpatialDimension
 
+// DimensionExclusionsSet represents set of excluded dimensions.
+type DimensionExclusionsSet []string
+
 // DimensionsSpec is responsible for configuring Druid's dimensions. They're a
 // set of columns in Druid's data model that can be used for grouping, filtering
 // or applying aggregations.
 // https://druid.apache.org/docs/latest/ingestion/ingestion-spec#dimensionsspec
 type DimensionsSpec struct {
-	Dimensions           DimensionSet        `json:"dimensions,omitempty"`
-	DimansionExclusions  DimensionSet        `json:"dimansionExclusions,omitempty"`
-	SpatialDimensions    SpatialDimensionSet `json:"spatialDimensions,omitempty"`
-	IncludeAllDimensions bool                `json:"includeAllDimensions,omitempty"`
-	UseSchemaDiscovery   bool                `json:"useSchemaDiscovery,omitempty"`
+	Dimensions           DimensionSet           `json:"dimensions,omitempty"`
+	DimensionExclusions  DimensionExclusionsSet `json:"dimensionExclusions,omitempty"`
+	SpatialDimensions    SpatialDimensionSet    `json:"spatialDimensions,omitempty"`
+	IncludeAllDimensions bool                   `json:"includeAllDimensions,omitempty"`
+	UseSchemaDiscovery   bool                   `json:"useSchemaDiscovery,omitempty"`
+}
+type QueryGranularity struct {
+	Type string `json:"type,omitempty"`
 }
```

Comment:
```suggestion

type QueryGranularity struct {
	Type string `json:"type,omitempty"`
}
```

And add a comment please.